gke/deployment.yaml

```diff
@@ -0,0 +1,54 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: clip-search
+  name: clip-search
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: clip-search
+  template:
+    metadata:
+      labels:
+        app: clip-search
+    spec:
+      containers:
+      # image registry URI should be replaced once GitHub Action is set up
+      - image: gcr.io/titanium-vision-318200/search_service
+        name: search-service-p4sml
+        imagePullPolicy: Always
+        ports:
+          - containerPort: 8080
+            protocol: TCP
+        livenessProbe:
+          httpGet:
+            path: /alive
+            port: 8080
+          initialDelaySeconds: 90
+          periodSeconds: 10
+        readinessProbe:
+          httpGet:
+            path: /ready
+            port: 8080
+          initialDelaySeconds: 90
+          periodSeconds: 10
+        resources: 
+          limits:
+            cpu: 1500m
+            memory: 4G
+---
+apiVersion: autoscaling/v1
+kind: HorizontalPodAutoscaler
+metadata:
+ name: clip-search-hpa
+
+spec:
+ scaleTargetRef:
+   apiVersion: apps/v1
+   kind: Deployment
+   name: clip-search
+ minReplicas: 1
+ maxReplicas: 3
+ targetCPUUtilizationPercentage: 80
```

Comment:
I am a bit confused. It appears to me that this YAML file is essentially doing what I had specified in [this shell script](https://github.com/deep-diver/image_search_with_natural_language/blob/main/server/k8s_deploy.sh) barring a few exceptions. Is there any reason we are going with YAML files instead of commands? Could you expand on this a bit more?

Reply:
AFAIK, `kubectl` CLI doesn't come with a full featured configurationality. There are lots of stuffs that can only be configured via `YAML` spec. 

Also, It is more readable and reproducible. All we have to do is `kubectl apply -f ...yaml`.

Reply:
Ok cool.